NetSDK/Services/Client/Classes/SelfRefreshingClient.cs

```diff
@@ -78,6 +87,9 @@ private void ReadConfig()
             RandomizeRefreshRates = bool.Parse(ConfigurationManager.AppSettings["RANDOMIZE_REFRESH_RATE"]);
             BlockMilisecondsUntilReady = int.Parse(ConfigurationManager.AppSettings["BLOCK_MILISECONDS_UNTIL_READY"]);
             ConcurrencyLevel = int.Parse(ConfigurationManager.AppSettings["SPLITS_STORAGE_CONCURRENCY_LEVEL"]);
+            TreatmentLogRefreshRate = int.Parse(ConfigurationManager.AppSettings["TEST_IMPRESSIONS_REFRESH_RATE"]);
+            TreatmentLogCacheSizePerTest = int.Parse(ConfigurationManager.AppSettings["TEST_IMPRESSIONS_CACHE_SIZE_PER_TEST"]);
```

Comment:
I haven't checked this part, but once again, consider an AtomicBlockingList of KeyImpressions.